app/components/TypewriterText.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect, useState } from "react";
+
+interface TypewriterTextProps {
+  text: string;
+  speed?: number;
+  render?: (text: string) => React.ReactNode;
+  onDone?: () => void;
+}
+
+export default function TypewriterText({ text, speed = 15, render, onDone }: TypewriterTextProps) {
```

Comment:
```suggestion
export default function TypewriterText({ text, speed = 25, render, onDone }: TypewriterTextProps) {
```
I think a slightly faster speed feels more natural, and smoother.

Reply:
Agreed. I played with making it variable (based on message length) but it was getting overly complex...